components/PlantForm/PlantForm.js

```diff
@@ -143,13 +220,24 @@ export default function PlantForm({
                 id={option.id}
                 name="lightNeed"
                 value={option.value}
-                required={option.id === "lightNeed1"}
                 defaultChecked={initialData?.lightNeed === option.value}
```

Comment:
This is a suggestion to reduce the amount of code neccessary to handle the custom error validation in your component: 

Consider changing `defaultChecked` so that
- the first in three radio buttons in a group is has a prop `defaultChecked`
- the second and third radio button have the value of the plant which is currently edited, if in edit-mode, for Example: `defaultChecked={initialData?.lightNeed === option.value}`.

As a result, your form would always have one radio button checked, regardless of wether the user is adding or editing a plant. Positive side effects would be improved UX and less code needed for error validation.

These suggestions apply to both radio button groups.

Of course this depends on how you want the user to use your form. If you prefer to have the user making an input in every section of the form, please feel free to stick to your logic as it is solid.